### PR TITLE
[Wasm GC] Full precompute support for GC

### PR DESCRIPTION
**Pull request by kripken**
The precompute pass ignored all reference types, but that was overly
pessimistic: we can precompute some of them, namely a null and a
reference to a function are fully precomputable, etc.

To allow that to work, add missing integration in `getFallthrough` as well.

With this, we can precompute quite a lot of field accesses in the existing
`-Oz` testcase, as can be seen from the output. That testcase runs
`--fuzz-exec` so it prints out all those logged values, proving they have
not changed.

### CODE REVIEW

File: test/lit/passes/precompute-gc.wast [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (ref.null func)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (i32.const 1)
+ ;; CHECK-NEXT: )
+ (func $test-fallthrough (result i32)
+  (local $x funcref)
+  (local.set $x
+   ;; the fallthrough value should be used (which is possible as with names
```
Comment (tlively) [inline test/lit/passes/precompute-gc.wast:21]: This comment looks cut off.

**Review by tlively — APPROVED** [review]

File: test/lit/passes/precompute-gc.wast [hunk 2]
```diff
@@ -0,0 +1,35 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: wasm-opt %s --remove-unused-names --precompute-propagate -all -S -o - \
```
Comment (tlively) [inline test/lit/passes/precompute-gc.wast:2]: Why `--remove-unused-names`?

File: test/lit/passes/precompute-gc.wast (hunk 1 above)
Comment (kripken) [reply test/lit/passes/precompute-gc.wast:21]: Oh, heh, that's what was explaining `--remove-unused-names`, fixing...

**Review by tlively — APPROVED** [review]
Makes sense!

**Comment by kripken** [comment]
Fuzzed for a few hours, looks ok...

**Pull request merged**